src/mocks/utils.ts

```diff
@@ -15,34 +16,50 @@ import {
 import { LoggedInUser } from "types/user"
 
 const apiDataBuilder = <T extends DefaultBodyType = DefaultBodyType>(
-  endpoint: string
+  endpoint: string,
+  requestType: "get" | "post"
 ) => (
   mockData: T,
   delay?: number | "infinite"
-): ReturnType<typeof rest["get"]> => {
-  return rest.get(endpoint, (req, res, ctx) => {
-    return res(delay ? ctx.delay(delay) : ctx.delay(0), ctx.json(mockData))
-  })
+): ReturnType<typeof rest["get"] | typeof rest["post"]> => {
+  switch (requestType) {
+    case "post":
+      return rest.post(endpoint, (req, res, ctx) => {
+        return res(delay ? ctx.delay(delay) : ctx.delay(0), ctx.json(mockData))
+      })
+    case "get":
+      return rest.get(endpoint, (req, res, ctx) => {
+        return res(delay ? ctx.delay(delay) : ctx.delay(0), ctx.json(mockData))
+      })
+    default:
+      return rest.get(endpoint, (req, res, ctx) => {
+        return res(delay ? ctx.delay(delay) : ctx.delay(0), ctx.json(mockData))
+      })
```

Comment:
i think we might have more http verbs in the future and we could instead extend this to be `rest[verb](...)`, if we can restrict the verb to `"get" | "post" | "delete"` etc and set a default to `get`. wdyt about this?

Reply:
That makes sense - changed in https://github.com/isomerpages/isomercms-frontend/pull/1093/commits/2cc56329ff030946f3f252bb1133a8d07992987b!